cmake/templates/OpenCVConfig.cmake.in

```diff
@@ -273,6 +273,22 @@ endif()
 # ==============================================================
 set(OpenCV_LIBRARIES ${OpenCV_LIBS})
 
+# Require C++11 for OpenCV modules
+if(CMAKE_VERSION VERSION_LESS "3.1")
+  if(NOT OpenCV_FIND_QUIETLY)
+    message(STATUS "OpenCV: CMake version is low (${CMAKE_VERSION}, required 3.1+). Can't enable C++11 features: https://github.com/opencv/opencv/issues/13000")
+  endif()
+else()
+  set(__target opencv_core)
+  if(TARGET opencv_world)
+    set(__target opencv_world)
+  endif()
+  # Simulate exported result of target_compile_features(opencv_core PUBLIC cxx_auto_type)
+  set_target_properties(${__target} PROPERTIES
+    INTERFACE_COMPILE_FEATURES "cxx_auto_type"
```

Comment:
I wouldn't list only `cxx_auto_type` but also other C++11 features actually used in OpenCV's interface (e.g., `cxx_rvalue_references` etc.) Optionally, one could replace these feature requirements by `cxx_std_11` given CMake version is 3.8 or above.

Reply:
Added CMake 3.8+ handling through `cxx_std_11`.

Extended list: `cxx_auto_type cxx_rvalue_references cxx_lambdas`
These features are mostly used in OpenCV public headers.

Reply:
LGTM.